autoload/rails.vim

```diff
@@ -4151,6 +4151,21 @@ function! s:url_encode(str, ...) abort
   return substitute(str, '[?@=&<>%#[:space:]' . (a:0 && a:1 == 'path' ? '' : ':/').']', '\=printf("%%%02X", char2nr(submatch(0)))', 'g')
 endfunction
 
+function! s:to_query(value, ...) abort
+  if exists('a:1')
+    let prefix = a:1
+  endif
+  if type(a:value) == v:t_dict
```

Comment:
Change these to `== type({})` style as I haven't pulled the plug on Vim 7 compatibility yet.

Reply:
Sounds good 👍  In the midst of this, I realized it's actually the Rails connection adapter setting variables, and not MySQL using the connection string, which kinda puts a damper on things: https://github.com/rails/rails/blob/7b4efda97a0f3f57d713a7b3698c516c2d8e80ec/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L793 So it _does_ fix the error with nested keys, but doesn't do anything legitimate with them.

Reply:
Does `rails dbconsole` make use of the variables? If so, how?

Reply:
Looks like they're ignored, and `dbconsole` does it's own thing with `database.yml`: https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/railties/lib/rails/commands/dbconsole/dbconsole_command.rb#L22-L44